test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -56,11 +56,16 @@ var (
 	initStackManager sync.Once
 )
 
+// RetryStrategy is a function that given the current error and the number of retries, returns the type of retry to perform and a list of options to modify the configuration
+type RetryStrategy func(error, int) (RetryType, []GetStackOption)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
As this is a functor, would name it `RetryStrategyFn`